ultravox/training/train.py

```diff
@@ -294,14 +290,19 @@ def train(args: config_base.TrainConfig):
             lr_scheduler_type=args.lr_scheduler,
             warmup_steps=args.lr_warmup_steps,
             weight_decay=args.weight_decay,
-            fp16=dtype == torch.float16,
-            bf16=dtype == torch.bfloat16,
+            # fp16=dtype == torch.float16,
+            # bf16=dtype == torch.bfloat16,
```

Comment:
For some reason, when this is enabled, the `accelerate` library turns any non-FSDP weights to full precision which is not great for us. I could not find a way around this, but given that we could reproduce non-FSDP results exactly, I'm not worried about it for now.

I'll leave a TODO here to verify this on non-FSDP too.

Here's the piece of code that does this, including their explanation. If you can figure out why they're doing this, let me know 🙂 
<img width="820" alt="image" src="https://github.com/user-attachments/assets/47c11ea0-f455-458b-9d8e-24ee9a68b25f">

Reply:
Just verified that with all three PRs combined, the loss curve for non-FSDP workflow remains exactly the same. We The training speed also matches the previous version exactly.